src/ratingservice/main.py

```diff
@@ -13,103 +13,155 @@
 # limitations under the License.
 
 import os
-import json
-from flask import Flask,Response,jsonify,request
-from psycopg2 import pool
-
-connpool = None
-
-# connect to cloud sql
-def initConnection():
-    db_user = os.environ.get('CLOUD_SQL_USERNAME')
-    db_password = os.environ.get('CLOUD_SQL_PASSWORD')
-    db_name = os.environ.get('CLOUD_SQL_DATABASE_NAME')
-    db_connection_name = os.environ.get('CLOUD_SQL_CONNECTION_NAME')
-    host = '/cloudsql/{}'.format(db_connection_name)
-    db_config = {
-        'user': db_user,
-        'password': db_password,
-        'database': db_name,
-        'host': host
-    }
-    connpool = pool.ThreadedConnectionPool(minconn=1, maxconn=10, **db_config)
-    return connpool
-
-# read product ids
-def read_products():
-    res = []
-    with open('products.json') as f:
-        data = json.load(f)
-        for product in data['products']:
-            res.append(product['id'])
-    return res
-
-# create and populate table
-def populate_database():
+from flask import Flask, jsonify, request
+from psycopg2 import pool, DatabaseError, IntegrityError
+
+
+# If `entrypoint` is not defined in app.yaml, App Engine will look for an app
+# called `app` in `main.py`.
+db_connection_pool = None
+
+app = Flask(__name__)
+if (os.environ.get('DB_USERNAME') == None or os.environ.get('DB_PASSWORD') == None
+        or os.environ.get('DB_NAME') == None or os.environ.get('DB_HOST') == None):
+    print('error: environment vars DB_USERNAME, DB_PASSWORD, DB_NAME and DB_HOST must be defined.')
+    exit(1)
+
+
+def getConnection():
+    global db_connection_pool
+    if db_connection_pool == None:
+        cfg = {
+            'user': os.environ.get('DB_USERNAME'),
+            'password': os.environ.get('DB_PASSWORD'),
```

Comment:
Yes, but it will increase a complexity of the example. Regardless, in the following steps the CloudSQL instance will not have public IP which should seriously mitigate the risk of the getting access to the instance.